Libraries/CommonFunctions.psm1

```diff
@@ -2341,3 +2341,40 @@ Function Get-DeviceName
         -command "bash get_data_disk_dev_name.sh" -runAsSudo
     return $ret
 }
+
+Function Get-ExpectedCount {
+    param (
+        [String] $ip,
+        [String] $port,
+        [String] $username,
+        [String] $password,
+        [String] $size,
+        [String] $type
+    )
+    $vmCPUCount = Run-LinuxCmd -username $user -password $password -ip $ip -port $port -command "nproc" -ignoreLinuxExitCode
+    if ($vmCPUCount) {
+        Write-LogDbg "Successfully fetched nproc result: $vmCPUCount"
+    } else {
+        Write-LogErr "Could not fetch the nproc command result."
+    }
+    [int]$expectedCount = 0
+    switch ($type) {
+        {($_ -imatch "NVME")} {
```

Comment:
can be just like below, as it's more straight forward, and fitting your functionality design.
GPU {... ;break;}
NVME {...; break;}